howso/analysis_weights.amlg

```diff
@@ -204,6 +212,74 @@
 		weights_map
 	)
 
+	;helper function for the residuals computing methods used within Analysis (ConvergeResiduals and ConvergeIRW)
+	;estimates the number of samples necessary to allow for the correct amount of samples for a feature to be predicted
+	;across all of its features that share deviations
+	;parameters:
+	; desired_samples_per_feature: the number of samples desired for each feature
+	#!ComputeRequiredNumSamplesForSharedDeviationsWithTS
+	(let
+		(assoc
+			;map of the desired samples per feature multiplied by the number of features in each shared deviation group
+			desired_non_null_samples_per_group
+				(map
+					(lambda (* desired_samples_per_feature (size (current_value))) )
+					(keep !sharedDeviationGroupByPrimaryMap features)
+				)
+		)
+
+		(declare (assoc
+			sample_cases
+				(contained_entities [
+					(query_exists !internalLabelSession)
+					(if use_case_weights
+						(query_weighted_sample weight_feature 1000)
+						(query_sample 1000)
+					)
+				])
+		))
+
+		;take a sample of 1000 cases and get the average number of non_null values for features in each shared deviation group
+		(declare (assoc
+			avg_non_null_values_per_case_per_group
+				(map
+					(lambda
+						(floor (/
+							(apply "+"
+								(map
+									(lambda
+										;the number of non-null values in the case of features in the shared deviation group
+										(size (filter (retrieve_from_entity (current_value) (current_value 1))))
```

Comment:
(retrieve_from_entity) only works on a single case, so (size (filter (retrieve_from_entity... )) will always return either 1 or 0
I think you meant to put the (size (filter  outside of the map that's iterating over sample_cases ?